### ISSUE TITLE
Who fancies a rewrite?

### ISSUE THREAD

**Issue by ahmednuaman** [ahmednuaman/grunt-scss-lint#90]
It's Xmas time, mistletoe and wine and rewriting stuff. So, I'm thinking in the downtime it'll be a good idea to rewrite this module from the ground up and migrate the tests away from node-unit, so let's have a little survey:

1. Is it worth doing a rewrite?
2. I'll add coverage using Istanbul, is it then worth requiring new PRs be 100% covered?
3. I like Jasmine, but I also like Mocha, really they're pretty much both the same, what does everyone else think?
4. Is it worth adding support for different CI reporters (considering the gem just does junit, so I'll probably drop that bit of code)?

**Comment by ahmednuaman** [comment]
Ok, so I've rewritten the tests now in Mocha (https://travis-ci.org/ahmednuaman/grunt-scss-lint) and used Istanbul (https://codeclimate.com/github/ahmednuaman/grunt-scss-lint) for coverage too; time to refactor.

**Comment by thomasphorton** [comment]
:+1: 

things are a little crazy for me right now, but i'll help out as i can.

**Comment by tbremer** [comment]
@ahmednuaman & @thomasphorton so you know you'll like to tweak / rewrite the module?

Perhaps we can start a markdown or wiki page for a `1.0.0` or similarly versioned major release?

**Comment by tbremer** [comment]
Oh. Phone typing is hard... Sorry.

**Comment by thomasphorton** [comment]
Are we looking to add more features?

When I think of refactoring, I'm thinking about bringing things up to speed as far as modern design patterns. I haven't really taken stock of all of the pieces recently, so maybe making a 'freshness' list would help?

I don't have a ton of time, but once the emails start flying around I can't help but get excited :smile:

**Comment by tbremer** [comment]
I actually got some dev cycles removed from my plate today. :+1: (which doesn't mean that tomorrow morning they won't get filled right back up).

I'd personally would like to see the events emitted cleaned up a bit (this is also something I added in, haha). Perhaps @ahmednuaman can speak to what changes he'd like to see as well.

1. events (_both error and success combined into a single event_)
1. xml parsing (_I remember reading in one of the issues freshening this up_)
